releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java

```diff
@@ -72,15 +72,24 @@ private MainProcess() {
      *
      * @param releaseNotes map of release notes messages.
      * @param cliOptions command line options.
+     * @param hasEncounteredErrorsBefore if any errors have occured before this method is called
      * @return list of publication errors.
      * @throws IOException if I/O error occurs.
      * @throws TemplateException if an error occurs while generating freemarker template.
      */
     public static List<String> runPostGenerationAndPublication(
-            Multimap<String, ReleaseNotesMessage> releaseNotes, CliOptions cliOptions)
+            Multimap<String, ReleaseNotesMessage> releaseNotes, CliOptions cliOptions,
+            boolean hasEncounteredErrorsBefore)
             throws IOException, TemplateException {
         runPostGeneration(releaseNotes, cliOptions);
-        return runPostPublication(cliOptions);
+        final List<String> errors = new ArrayList<>();
+        if (cliOptions.isValidateVersion()) {
+            errors.addAll(validateNotes(releaseNotes, cliOptions));
+        }
+        if (!hasEncounteredErrorsBefore) {
```

Comment:
Rename to shouldRunPublication

Reply:
Done. Also added the check for `errors.isEmpty()` we forgot.